lib/google_calendar.rb

```diff
@@ -56,14 +61,18 @@ def events_as(who, date)
     @logger.info("Attempting to receive events for "+who)
     @logger.debug details.to_yaml
 
-    ret = @client.execute(
-      api_method: @calendar.events.list,
-      parameters: {'calendarId' => who, 'sendNotifications' => true},
-      body: details.to_json,
-      headers: {'Content-Type' => 'application/json'}
-    )
+    ret = @calendar.list_events(
+        who
+      )
 
     @logger.debug ret.to_yaml
-    ret    
+    ret.to_h  
+  end
+
+  def cleanup!
```

Comment:
I meant you could rewrite it like this:
```ruby
incoming_events.each do |event|
  GoogleCalendar.open(interpolate_options(options, event), Rails.logger) do |calendar|
    cal_message = event.payload["message"]
    # ...
  end
end
```
This way the caller (GoogleCalendarPublishAgent#receive) doesn't need to bother with `cleanup!`, and the callee (GoogleCalendar.open) can easily manage thread safety because it takes a block in which a whole session is enclosed.